extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/WorkerImpl.java

```diff
@@ -260,13 +264,36 @@ public Optional<MSQErrorReport> runTask(final Closer closer) throws Exception
       }
     });
 
+    long maxAllowedParseExceptions = Long.parseLong(task.getContext().getOrDefault(
+        MSQWarnings.CTX_MAX_PARSE_EXCEPTIONS_ALLOWED,
+        Long.MAX_VALUE
+    ).toString());
+
+    long maxVerboseParseExceptions;
+    if (maxAllowedParseExceptions == -1L) {
+      maxVerboseParseExceptions = Limits.MAX_VERBOSE_PARSE_EXCEPTIONS;
+    } else {
+      maxVerboseParseExceptions = Math.min(maxAllowedParseExceptions, Limits.MAX_VERBOSE_PARSE_EXCEPTIONS);
+    }
+
+    Set<String> disallowedWarningCode = ImmutableSet.of();
```

Comment:
Nit: 
you can just set this to null and initialize it in the else part. 
Also, we can rename this to critical error codes